Bebop front camera subsampling

wilcoschoneveld opened the pull request:
This pull request allows for sub-sampling in the bebop front camera to reduce the camera resolution and increase performance. I took the work from @wv-tud and updated it with the latest master. The current settings give a steady 15 fps.

I'm trying to figure out how to correctly combine subsampling with a desired output size in order to get a more conventional image size, but there are a number of programming restrictions to follow; http://www.onsemi.com/pub_link/Collateral/MT9F002-D.PDF

kirkscheper commented on sw/airborne/boards/bebop/mt9f002.c:
```diff
@@ -781,6 +781,52 @@ static void mt9f002_set_gains(struct mt9f002_t *mt)
+	    write_reg(mt, MT9F002_SCALING_MODE, 2, 2); // Vertical and horizontal scaling
+	    write_reg(mt, MT9F002_SCALE_M, scaleFactor, 2);
+	  }
+	  printf("OUTPUT_SIZE: (%i, %i)\tSCALED_SIZE: (%i, %i)\n", mt->output_width, mt->output_height, mt->scaled_width, mt->scaled_height);
```
I guess this was for debugging?

wilcoschoneveld replied:
I guess it's helpful and not intrusive, or is this against code guidelines?

kirkscheper replied:
The other debugging printouts have a specific format so I assume you should follow that.

wilcoschoneveld replied:
I changed the printout, how can we close this review?

wv-tud commented:
Great work Wilco! Thanks for the pull request

@kirk that was indeed for debugging, the resolution/scalar setting are not very intuitive when using subsampling so maybe some resolution feedback in a different way is nice though...

flixr commented:
@fvantienen mentioned that there might be problems including ISP code in Paparazzi... are these latest additions affected?

wv-tud commented:
@flixr I am not a 100% on this but I am pretty sure the "problem code" is
the proprietary H264 encoding part which is not included in this commit

On Mon, Sep 26, 2016, 21:21 Felix Ruess <notifications@github.com> wrote:

> @fvantienen <https://github.com/fvantienen> mentioned that there might be
> problems including ISP code in Paparazzi... are these latest additions
> affected?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/paparazzi/paparazzi/pull/1856#issuecomment-249670249>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQuu9pL_EK7XyAvGUE0IrWJYyAGpaoahks5quBtGgaJpZM4J-ztR>
> .
>

wilcoschoneveld commented:
I updated the bebop example with a 640 x 640 front camera output and a reasonably steady frame-rate (26-30 fps):

![front_cam](https://cloud.githubusercontent.com/assets/3064588/18987103/67b8aaa2-8700-11e6-8391-fb880357c47a.png)

I also rebased on top of master.

kirkscheper commented:
@wilcoschoneveld  Check the compile errors from travis

wilcoschoneveld commented:
replaced by #1908
